### ISSUE TITLE
[Question/Help]  Taiga keeps redirecting to `/discover`

### ISSUE THREAD

**Issue by Minkiu** [taigaio/taiga-docker#50]
Hello there,

I am trying to set up taiga to try it, I am following the documentation (I checked out the `stable` branch), but whenever I try to access it, it redirects me to `host/discover` and none of the containers seem to be erroring afaik, my first question is:

1. I need to run a nginx (or apache, or traefik...) on top of the taiga-dockers, the one where I actually define my server name, no?

As per: https://taigaio.github.io/taiga-doc/dist/setup-production.html#_configure_the_proxy

If the answer is yes, the second question is:

2. Is the above example incomplete?
 (that's why the `...`?) I should map the same endpoints as there are in: https://github.com/taigaio/taiga-docker/blob/master/taiga-gateway/taiga.conf ?

I cloned the repo, and this are all the changes I've made (i will change to proper ones later and change teh passwords, I just want to get it to run first):
```
diff --git a/docker-compose.yml b/docker-compose.yml
index 0c5b18e..849581e 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -24,8 +24,13 @@ x-environment:
   # Should be the same as in taiga-async-rabbitmq and taiga-events-rabbitmq
   RABBITMQ_USER: taiga
   RABBITMQ_PASS: taiga
+  EVENTS_PUSH_BACKEND_URL: "amqp://taiga:taiga@taiga-events-rabbitmq:5672/taiga"
+  CELERY_BROKER_URL: "amqp://taiga:taiga@taiga-async-rabbitmq:5672/taiga"
   # Telemetry settings
-  ENABLE_TELEMETRY: "True"
+  ENABLE_TELEMETRY: "False"
+  SESSION_COOKIE_SECURE: "False"
+  CSRF_COOKIE_SECURE: "False"
+  PUBLIC_REGISTER_ENABLED: "True"
 
 x-volumes:
   &default-back-volumes
@@ -84,6 +89,7 @@ services:
     environment:
       TAIGA_URL: "http://localhost:9000"
       TAIGA_WEBSOCKETS_URL: "ws://localhost:9000"
+      PUBLIC_REGISTER_ENABLED: "True"
     networks:
       - taiga
     # volumes:
```

I tried mapping (i.e duplicate the taiga.conf from the gateway and change all the urls to `localhost:9000`) with the same result, any idea what I might be doing wrong?

Cheers.

**Comment by yamila-moreno** [comment]
:wave: 

As I can see, the second  `PUBLIC_REGISTER_ENABLED` (in taiga-front) is wrong, it should be `true` (as in JS) instead of `True` (as in Python).
You can quick check removing this variables (both in x-environment and taiga-front).

Let me know if this works!

Cheers!

**Comment by Minkiu** [comment]
Hey, thanks for the quick reply, I tired removing the env vars `PUBLIC_REGISTER_ENABLED`, also tried setting the `taiga-front` to the correct `true` (then in each case did `docker-compose down` just to be on the safe) and I start it by launching the `./launch-taiga.sh`, and the same outcome happens, the `/discover` page, shows the following:
```
Taiga

Something happened and the Taiga has captured the error to be able to work on it.
Take me home
```

The `Take me home` link is "working" but I end up redirected in the discover page, with the above message, where is this "captured error" lives?

I tried accessing `/api` and `/admin` and I get an "Not Found" error, so I think the problem might be on my Nginx conf? I have it as follow (I) redacted the actual `server_name`), this is *not* the one in the `taiga-gateway/taiga.conf`:
```
server {
  server_name my.example.domain.com;

  location / {
    proxy_set_header Host $http_host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Scheme $scheme;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_redirect off;
    proxy_pass http://localhost:9000/;
  }
    # Api
    location /api {
        proxy_pass http://localhost:9000/api;
        proxy_pass_header Server;
        proxy_set_header Host $http_host;
        proxy_redirect off;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Scheme $scheme;
    }

    # Admin
    location /admin {
        proxy_pass http://localhost:9000/admin;
        proxy_pass_header Server;
        proxy_set_header Host $http_host;
        proxy_redirect off;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Scheme $scheme;
    }

    # Static
    location /static {
        root /taiga;
    }

    # Media
    location /_protected {
        internal;
        alias /taiga/media/;
        add_header Content-disposition "attachment";
    }

    # Unprotected section
    location /media/exports {
        alias /taiga/media/exports/;
        add_header Content-disposition "attachment";
    }

    location /media {
        proxy_set_header Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Scheme $scheme;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_pass http://localhost:9003/;
        proxy_redirect off;
    }


  # Events
  location /events {
      proxy_pass http://localhost:9000/events;
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";
      proxy_connect_timeout 7d;
      proxy_send_timeout 7d;
      proxy_read_timeout 7d;
  }
}
```

Cheers.

**Comment by yamila-moreno** [comment]
> Hey, thanks for the quick reply, I tired removing the env vars `PUBLIC_REGISTER_ENABLED`, also tried setting the `taiga-front` to the correct `true` (then in each case did `docker-compose down` just to be on the safe) and I start it by launching the `./launch-taiga.sh`, and the same outcome happens, the `/discover` page, shows the following:
> 
> ```
> Taiga
> 
> Something happened and the Taiga has captured the error to be able to work on it.
> Take me home
> ```

`docker-compose ps` shows everything up and running?
did you check where is the api calling? (in dev tools)

> The `Take me home` link is "working" but I end up redirected in the discover page, with the above message, where is this "captured error" lives?
> 
> I tried accessing `/api` and `/admin` and I get an "Not Found" error, 

urls are `/api/` and `/admin/` (with the final slash)

> so I think the problem might be on my Nginx conf? I have it as follow (I redacted the actual `server_name`), this is _not_ the one in the `taiga-gateway/taiga.conf`:

if you're using `taiga-docker` repository, you don't need to map anything else in your nginx apart from the example, in your host nginx, only `/` and `/events`
Thus, the configuration _in your nginx_ should be something like https://taigaio.github.io/taiga-doc/dist/setup-production.html#_configure_the_proxy

```
server {
  server_name taiga.mycompany.com;

  # logs, tls, etc

  location / {
    proxy_set_header Host $http_host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Scheme $scheme;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_redirect off;
    proxy_pass http://localhost:9000/;
  }

  # Events
  location /events {
      proxy_pass http://localhost:9000/events;
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";
      proxy_connect_timeout 7d;
      proxy_send_timeout 7d;
      proxy_read_timeout 7d;
  }
}
```
Let's check:
- [ ] you've the latest version of taiga-docker
- [ ] the latest images (pull them again if not sure, we just uploaded a new one)
- [ ] how are you launching this?
- [ ] which url are you accessing (the one that gives you error)?
- [ ] you should be able to access to `http://localhost:9000`
- [ ] can you check in the dev tools where is the API pointing?

**Comment by Minkiu** [comment]
Hey,

so I sorted it, thanks for the help and patience, 

the problem was that I am setting this up in a VM and trying to access it from another computer, after you pointed out to check the dev tools (I should have looked at this to begin with... so sorry) it was still pointing to `http://localhost:9000` so I went into the `docker-compose.yml` and changed every instance of `localhost:9000` to my domain, and now it works: `TAIGA_SITES_DOMAIN`, `TAIGA_URL` and `TAIGA_WEBSOCKETS_URL`.

I was under the impression (my ignorance), that since there was a nginx on top of the whole `taiga-docker` those env vars could stay as `localhost:9000` and it would still work...

So yeah, sorry again.

Thank you.
